src/Toolkit/src/Compiler/TwigComponentCompiler.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Toolkit\Compiler;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\UX\Toolkit\Compiler\Exception\TwigComponentAlreadyExist;
+use Symfony\UX\Toolkit\Registry\DependenciesResolver;
+use Symfony\UX\Toolkit\Registry\Registry;
+use Symfony\UX\Toolkit\Registry\RegistryItem;
+use Symfony\UX\Toolkit\Registry\RegistryItemType;
+
+/**
+ * @author Jean-François Lépine
+ *
+ * @internal
+ */
+final class TwigComponentCompiler
+{
+    public function __construct(
+        private readonly ?string $prefix,
+        private readonly DependenciesResolver $dependenciesResolver,
+    ) {
+    }
+
+    public function compile(
+        Registry $registry,
+        RegistryItem $item,
+        string $directory,
+        bool $overwrite = false,
+    ): void {
+        // resolve dependencies (avoid circular dependencies)
+        $this->dependenciesResolver->resolve($registry);
+
+        $filesystem = new Filesystem();
+        if (!$filesystem->exists($directory)) {
+            $filesystem->mkdir($directory);
+        }
+
+        // We need to install all children components of each parent component
+        $componentsToInstall = [];
+        foreach($item->getParents() as $parentName) {
+            // Children of the parent component
+            $componentsToInstall = array_merge($componentsToInstall, $registry->get($parentName)->getChildren());
+
+            // And the parent itself
+            $componentsToInstall = array_merge([$parentName], $componentsToInstall);
```

Comment:
Could it be simplified to this?
```suggestion
            $componentsToInstall = [
                $parentName, 
                ...$componentsToInstall, 
                // children of the parent component
                ...$registry->get($parentName)->getChildren(),
            );
```